src/test/java/com/amplitude/AmplitudeTest.java

```diff
@@ -335,6 +339,27 @@ public void testWithSwallowMiddleware()
     assertEquals("middleware_user_2", sentEvent.userId);
   }
 
+  @Test
+  public void testShouldWait() throws NoSuchFieldException, IllegalAccessException {
+    Amplitude amplitude = Amplitude.getInstance();
+    amplitude.init(apiKey);
+    Event event = new Event("test event", "test-user-0");
+    Event event2 = new Event("test event", "test-user-1");
+    assertFalse(amplitude.shouldWait(event));
+    Field httpTransportField = amplitude.getClass().getDeclaredField("httpTransport");
+    httpTransportField.setAccessible(true);
+    HttpTransport transport = (HttpTransport) httpTransportField.get(amplitude);
+    amplitude.setRecordThrottledId(true);
```

Comment:
Do we need to set this setting first?

Reply:
Yes amplitude.setRecordThrottledId(true) is a config setting to record ids with 429 response and use the ids for shouldWait() method

Reply:
The first assertFalse I put their on purpose. It should be false before set the setRecordThrottledId config. I added another assertFalse after config.